geomesa-process/geomesa-process-vector/src/main/scala/org/locationtech/geomesa/process/analytic/HeatmapSurface.scala

```diff
@@ -0,0 +1,119 @@
+/***********************************************************************
+ * Copyright (c) 2013-2022 Commonwealth Computer Research, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, Version 2.0
+ * which accompanies this distribution and is available at
+ * http://www.opensource.org/licenses/apache2.0.php.
+ ***********************************************************************/
+
+package org.locationtech.geomesa.process.analytic
+
+import org.locationtech.jts.geom.Envelope
+
+class HeatmapSurface(kernelRadius: Int,
```

Comment:
@elahrvivaz I basically just ripped this from geotools and added a ccri copyright header - not sure if this is okay or not. Unfortunately I had to copy the whole class because java doesn't allow overriding private methods. Thoughts?

Reply:
I confirmed with @jnh5y that we're not allowed to copy and re-license geotools as it's LGPL, so we'll need to come up with a different solution here.